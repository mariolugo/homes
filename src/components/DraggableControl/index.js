import React, { useState } from 'react';
import styled from 'styled-components';

const ControlStyled = styled.div`
  height: 35px;
  box-sizing: border-box;
  width: 290px;
  border-radius: 2px;
  background-color: rgb(255, 255, 255);
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;
  margin-bottom: 15px;
  cursor: pointer;
  position: absolute;
  bottom: 20px;
  left: 50%;
  margin-left: -148px;
  padding: 8px;
  span {
    font-size: 14px;
  }
`;

const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
  // Hide checkbox visually but remain accessible to screen readers.
  // Source: https://polished.js.org/docs/#hidevisually
  border: 0;
  clip: rect(0 0 0 0);
  clippath: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`;

const Icon = styled.svg`
  fill: none;
  stroke: white;
  stroke-width: 2px;
`;

const StyledCheckbox = styled.div`
  display: inline-block;
  width: 16px;
  height: 16px;
  border: 1px solid black;
  background: ${(props) => (props.checked ? 'rgb(141, 122, 255)' : 'white')};
  border-radius: 3px;
  transition: all 150ms;
  margin-right: 10px;
  ${HiddenCheckbox}:focus + & {
    box-shadow: 0 0 0 3px rgb(141, 122, 255);
  }
  ${Icon} {
    visibility: ${(props) => (props.checked ? 'visible' : 'hidden')};
  }
`;

const CheckboxContainer = styled.div`
  display: inline-block;
  vertical-align: middle;
`;

/**
 * Component used for the map, in here we have a checkbox and we can add logic to handle the checkbox.
 * For example we can activate the checkbox and filter the homes if the map changes
 */
const DraggableControl = () => {
  const [active, setActive] = useState(false);

  const onCheckboxChange = () => {
    setActive((prev) => !prev);
  };

  return (
    <ControlStyled>
      <div>
        <form>
          <label htmlFor="checkbox">
            <CheckboxContainer>
              <HiddenCheckbox id="checkbox" checked={active} onChange={onCheckboxChange} />
              <StyledCheckbox checked={active}>
                <Icon viewBox="0 0 24 24">
                  <polyline points="20 6 9 17 4 12" />
                </Icon>
              </StyledCheckbox>
            </CheckboxContainer>
            <span>Filtrar mientras te mueves en el mapa</span>
          </label>
        </form>
      </div>
    </ControlStyled>
  );
};

export default DraggableControl;
