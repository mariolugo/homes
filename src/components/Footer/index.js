import React from 'react';
import styled from 'styled-components';
import { Col, Row } from 'react-styled-flexboxgrid';
import { WrapperGrid } from '../';
import FooterLogo from '../../assets/icons/footer-logo.svg';
import Messenger from '../../assets/icons/messenger.svg';

const FooterStyled = styled.div`
  background-color: white;
  border-top: 1px solid ${({ theme }) => theme.colors.grayLight};
  margin: 0px auto;
  min-height: 60px;

  @media only screen and (max-width: 600px) {
    padding-top: 10px;
  }
`;

const LinkStyled = styled.p`
  font-size: 13px;
`;

const FooterLink = styled(LinkStyled)`
  text-align: center;
`;

const ListStyled = styled.ul`
  list-style: none;
  padding-left: 0px;
`;

const ListItemStyled = styled.li`
  svg {
    display: inline-block;
    margin-right: 5px;
    position: relative;
    top: 2px;
  }
  p {
    display: inline-block;
    margin: 0;
  }

  @media only screen and (max-width: 600px) {
    text-align: center;
  }
`;

const LogoContainer = styled(Col)`
  display: flex;
  padding-right: 0;
  align-items: center;
  justify-content: center;
`;

const ColCentered = styled(Col)`
  display: flex;
  align-items: center;
  justify-content: center;
`;

/**
 * Main footer of the application
 */
const Footer = () => (
  <FooterStyled>
    <WrapperGrid>
      <Row>
        <Col xs={12} sm={4}>
          <Row>
            <LogoContainer xs={12} sm={3}>
              <FooterLogo />
            </LogoContainer>
            <Col xs={12} sm={9}>
              <ListStyled>
                <ListItemStyled>
                  <LinkStyled>hey@homie.mx</LinkStyled>
                </ListItemStyled>
                <ListItemStyled>
                  <LinkStyled>5015 1546</LinkStyled>
                </ListItemStyled>
                <ListItemStyled>
                  <Messenger />
                  <LinkStyled>Escríbenos por facebook</LinkStyled>
                </ListItemStyled>
              </ListStyled>
            </Col>
          </Row>
        </Col>
        <Col xs={12} sm={3}>
          <ListStyled>
            <ListItemStyled>
              <LinkStyled>Te buscamos depa</LinkStyled>
            </ListItemStyled>
            <ListItemStyled>
              <LinkStyled>Blog</LinkStyled>
            </ListItemStyled>
            <ListItemStyled>
              <LinkStyled>Equipo</LinkStyled>
            </ListItemStyled>
          </ListStyled>
        </Col>
        <ColCentered xs={12} sm={2}>
          <FooterLink>Aviso de privacidad</FooterLink>
        </ColCentered>
        <ColCentered xs={12} sm={3}>
          <FooterLink>Términos y condiciones</FooterLink>
        </ColCentered>
      </Row>
    </WrapperGrid>
  </FooterStyled>
);

export default Footer;
