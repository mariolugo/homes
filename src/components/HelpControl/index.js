import React from 'react';
import styled from 'styled-components';
import Image from 'next/image';

const ControlStyled = styled.div`
  width: 50px;
  height: 50px;
  background-color: white;
  cursor: pointer;
  position: absolute;
  bottom: 20px;
  right: 50%;
  margin-right: 10px;
  box-sizing: border-box;
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;
  padding: 13px 9px;
  border-radius: 2px;
`;

/**
 * Help control that is required in the map.
 */
const HelpControl = () => {
  return (
    <ControlStyled>
      <Image src="/images/help_map.png" alt="Help Map" width={35} height={35} />
    </ControlStyled>
  );
};

export default HelpControl;
