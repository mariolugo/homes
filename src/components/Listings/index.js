import React from 'react';
import styled from 'styled-components';
import { Col, Row } from 'react-styled-flexboxgrid';
import { Post } from '../';

const ListingsContainerStyled = styled.div``;

const ListingsContainer = ({ posts = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15] }) => (
  <ListingsContainerStyled>
    <Row>
      {posts.length > 0 &&
        posts.map((p) => (
          <Col key={p} xs={12} md={6}>
            <Post />
          </Col>
        ))}
    </Row>
  </ListingsContainerStyled>
);

export default ListingsContainer;
