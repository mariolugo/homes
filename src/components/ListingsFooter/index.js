import React from 'react';
import styled from 'styled-components';

const ListingsFooterStyled = styled.div`
  p {
    color: ${({ theme }) => theme.colors.textPrimary};
  }
`;

const Title = styled.p`
  text-align: center;
  font-family: 'Open Sans Bold', sans-serif;
`;

const Subtitle = styled.p`
  text-align: center;
`;

const Button = styled.button`
  border-radius: 2px;
  border: none;
  width: 300px;
  display: block;
  height: 35px;
  background-color: ${({ theme }) => theme.colors.secondary};
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;
  color: white;
  font-family: 'Open Sans Bold', sans-serif;
  line-height: 19px;
  padding: 8px 0px;
  margin: 20px auto;
`;

const ListingsFooter = () => (
  <ListingsFooterStyled>
    <Title>¿No encuentras inmueble?</Title>
    <Subtitle>Dinos qué quieres y nosotros te lo buscamos</Subtitle>
    <Button>Encuéntrenme un inmueble</Button>
  </ListingsFooterStyled>
);

export default ListingsFooter;
