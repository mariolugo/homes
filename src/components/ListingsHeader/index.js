import React from 'react';
import styled from 'styled-components';
import { Col, Row } from 'react-styled-flexboxgrid';
import Filter from '../../assets/icons/filter.svg';

const ListingHeaderStyled = styled.div`
  width: 100%;
  margin-bottom: 15px;
`;

const FilterButton = styled.button`
  height: 50px;
  background-color: ${({ theme }) => theme.colors.primary};
  border: none;
  border-radius: 3px;
  color: white;
  width: 100%;
  cursor: pointer;
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;

  svg {
    display: inline-block;
    margin-right: 5px;
    position: relative;
    top: 2px;
  }

  p {
    display: inline-block;
    font-family: 'Open Sans Bold', sans-serif;
    font-size: 16px;
    margin: 0;
  }

  @media only screen and (max-width: 500px) {
    margin-top: 15px;
  }
`;

const InfoBox = styled.div`
  background-color: white;
  height: 50px;
  max-height: 50px;
  border-radius: 3px;
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;

  h3 {
    font-size: 16px;
    line-height: 1;
    padding: 7px 0px 0px;
    margin: 0px 0px 0px 10px;
    color: ${({ theme }) => theme.colors.primary};
    font-family: 'Open Sans Bold', sans-serif;
  }

  h4 {
    font-size: 14px;
    line-height: 1;
    padding: 7px 0px 0px;
    margin: 0px 0px 0px 10px;
    color: ${({ theme }) => theme.colors.textPrimary};
  }
`;

const ListingHeaders = () => (
  <ListingHeaderStyled>
    <Row>
      <Col xs={12} sm={9}>
        <InfoBox>
          <h3>Ciudad de Mexico</h3>
          <h4>230 departamentos</h4>
        </InfoBox>
      </Col>
      <Col xs={12} sm={3}>
        <FilterButton>
          <Filter />
          <p>Filtros</p>
        </FilterButton>
      </Col>
    </Row>
  </ListingHeaderStyled>
);

export default ListingHeaders;
