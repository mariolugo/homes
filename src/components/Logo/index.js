import React from 'react';
import Image from 'next/image';

/**
 * Main logo of the app
 */
const Logo = () => <Image src="/images/logo.png" alt="Logo" width={94} height={32} />;

export default Logo;
