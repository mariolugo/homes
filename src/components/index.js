import NavBar from './NavBar';
import Listings from './Listings';
import Map from './Map';
import Footer from './Footer';
import Post from './Post';
import WrapperGrid from './WrapperGrid';

export { NavBar, Listings, Map, Footer, Post, WrapperGrid };
