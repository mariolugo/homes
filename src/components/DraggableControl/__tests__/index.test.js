import { render } from '@testing-library/react';
import { DraggableControl } from '../../../components';

/** @test {DraggableControl Component} */
describe('DraggableControl Component', () => {
  it('should render without crashing', () => {
    const wrapper = render(<DraggableControl />);
    expect(wrapper).toMatchSnapshot();
  });
});
