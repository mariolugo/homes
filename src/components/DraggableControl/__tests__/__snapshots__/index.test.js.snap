// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableControl Component should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-eggNIi foBzdd"
      >
        <div>
          <form>
            <label
              for="checkbox"
            >
              <div
                class="sc-bBrOnJ jzNVSf"
              >
                <input
                  class="sc-cTkwdZ husymC"
                  id="checkbox"
                  type="checkbox"
                />
                <div
                  class="sc-dOSReg fCWMZY"
                >
                  <svg
                    class="sc-jNMdTA hmbXfa"
                    viewBox="0 0 24 24"
                  >
                    <polyline
                      points="20 6 9 17 4 12"
                    />
                  </svg>
                </div>
              </div>
              <span>
                Filtrar mientras te mueves en el mapa
              </span>
            </label>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-eggNIi foBzdd"
    >
      <div>
        <form>
          <label
            for="checkbox"
          >
            <div
              class="sc-bBrOnJ jzNVSf"
            >
              <input
                class="sc-cTkwdZ husymC"
                id="checkbox"
                type="checkbox"
              />
              <div
                class="sc-dOSReg fCWMZY"
              >
                <svg
                  class="sc-jNMdTA hmbXfa"
                  viewBox="0 0 24 24"
                >
                  <polyline
                    points="20 6 9 17 4 12"
                  />
                </svg>
              </div>
            </div>
            <span>
              Filtrar mientras te mueves en el mapa
            </span>
          </label>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
